pkg/controller/jobs/noop/noop_controller.go

```diff
@@ -0,0 +1,40 @@
+package noop
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+)
+
+var (
+	_   jobframework.JobReconcilerInterface = (*Reconciler)(nil)
+	gvk                                     = corev1.SchemeGroupVersion.WithKind("Noop")
+)
+
+type Reconciler struct {
+}
+
+func (r Reconciler) Reconcile(_ context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	return ctrl.Result{}, nil
+}
+
+func (r Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	concurrency := mgr.GetControllerOptions().GroupKindConcurrency[gvk.GroupKind().String()]
+	ctrl.Log.V(3).Info("Setting up Noop reconciler", "concurrency", concurrency)
```

Comment:
The concurrency of doing nothing is not very relevant. We can make the `Reconciler` hold some kind of info about the type for which the noop reconciler is "started" and log something like 

`"Setting up  Noop reconciler for %q ", r.gvk `